src/dftbp/dftb/hybridxc.F90

```diff
@@ -3563,6 +3814,121 @@ function getCamGammaGSum(this, iAt1, iAt2, iSp1, iSp2, rCellVecsG) result(gamma)
   end function getCamGammaGSum
 
 
+  !> Calculates pseudo Fourier transform of square CAM y-matrix with shape [nOrb, nOrb].
+  subroutine getCamGammaFourierSqr(this, iSquare, cellVecsG, rCellVecsG, kPoint, kPointPrime,&
+      & camGammaSqr)
+
+    !> Class instance
+    class(THybridXcFunc), intent(in) :: this
+
+    !> Position of each atom in the rows/columns of the square matrices. Shape: (nAtom)
+    integer, intent(in) :: iSquare(:)
+
+    !> Translation vectors to lattice cells in units of lattice constants
+    real(dp), intent(in) :: cellVecsG(:,:)
+
+    !> Vectors to unit cells in absolute units
+    real(dp), intent(in) :: rCellVecsG(:,:)
+
+    !> First and second k-point in relative coordinates
+    real(dp), intent(in) :: kPoint(:), kPointPrime(:)
+
+    !> Pseudo Fourier transform of square gamma matrix
+    complex(dp), intent(out) :: camGammaSqr(:,:)
+
+    !! Indices of shifted atom
+    integer :: iAt1, iSp1
+
+    !! Indices of atom in central cell
+    integer :: iAt2, iSp2
+
+    !! Number of atoms in central cell
+    integer :: nAtom0
+
+    !! Auxiliary variables
+    integer :: iOrbStart1, iOrbEnd1, iOrbStart2, iOrbEnd2
+
+    nAtom0 = size(this%species0)
+
+    camGammaSqr(:,:) = (0.0_dp, 0.0_dp)
+
+    do iAt2 = 1, nAtom0
+      iSp2 = this%species0(iAt2)
+      iOrbStart2 = iSquare(iAt2)
+      iOrbEnd2 = iSquare(iAt2 + 1) - 1
+      do iAt1 = 1, nAtom0
+        iSp1 = this%species0(iAt1)
+        iOrbStart1 = iSquare(iAt1)
+        iOrbEnd1 = iSquare(iAt1 + 1) - 1
+        camGammaSqr(iOrbStart1:iOrbEnd1, iOrbStart2:iOrbEnd2)&
+            & = getCamGammaFourier(this, iAt1, iAt2, iSp1, iSp2, cellVecsG, rCellVecsG, kPoint,&
+            & kPointPrime)
+      end do
+    end do
+
+  end subroutine getCamGammaFourierSqr
+
+
+  !> Returns pseudo Fourier transform of long-range and full-range Hartree-Fock gamma's.
+  function getCamGammaFourier(this, iAt1, iAt2, iSp1, iSp2, cellVecsG, rCellVecsG, kPoint,&
+      & kPointPrime) result(gamma)
+
+    !> Class instance
+    class(THybridXcFunc), intent(in) :: this
+
+    !> Index of first and second atom
+    integer, intent(in) :: iAt1, iAt2
+
+    !> Index of first and second species
+    integer, intent(in) :: iSp1, iSp2
+
+    !> Translation vectors to lattice cells in units of lattice constants
+    real(dp), intent(in) :: cellVecsG(:,:)
+
+    !> Vectors to unit cells in absolute units
+    real(dp), intent(in) :: rCellVecsG(:,:)
+
+    !> First and second k-point in relative coordinates
+    real(dp), intent(in) :: kPoint(:), kPointPrime(:)
+
+    !> Resulting Gamma, summed up for g-vectors
+    complex(dp) :: gamma
+
+    !! Phase factor
+    complex(dp) :: phase
+
+    !! Distance between the two atoms
+    real(dp) :: dist
+
+    !! k-point in relative coordinates multiplied by 2pi
```

Comment:
Is the comment correct, as you store the shift between k and k' in this (x 2pi)

Reply:
I changed it to "Difference (k - k') with both k-points in relative coordinates, multiplied by 2pi" to be on the safe side. Also, it kind of violated the style-guide, although no one said that the capitalization also holds for internal variables ;)